src/extrapolation/filled.jl

```diff
@@ -54,12 +54,13 @@ end
     return ret
 end
 
-@inline function Interpolations.gradient(etp::FilledExtrapolation{T}, x::Vararg{Number,N}) where {T,N}
+@inline function Interpolations.gradient(etp::FilledExtrapolation{T, N, ITP}, x::Vararg{Number, N}) where {T,N,ITP}
     itp = parent(etp)
+    check_type() = Base._return_type(gradient, Tuple{typeof(itp), map(typeof, x)...})
     if checkbounds(Bool, itp, x...)
         gradient(itp, x...)
     else
-        ptype = promote_type(eltype(itp), map(typeof, x)...)
+        ptype = eltype(check_type())
         return zeros(SVector{ndims(etp), ptype})
```

Comment:
Actually, I don't know why wrapping as a function makes it type stable. Any ideas?

---

```suggestion
        return zeros(check_type())
```

Reply:
[This](https://discourse.julialang.org/t/could-base-return-type-ever-be-inferable/8057) thread did shed some insight as to how to make `Base._return_type` inferable, but I still don't quite firmly understand how did that make it stable.